### PR TITLE
Fluids Conditions do not work #2342

### PR DESCRIPTION
**Pull request by Dagst**

### CODE REVIEW

File: data/actions/scripts/other/fluids.lua [hunk 1]
```diff
@@ -9,12 +9,22 @@ poison:setParameter(CONDITION_PARAM_STARTVALUE, -5)
...
+	[15] = "Aah...",
+	[19] = "Urgh!",
+	[43] = "Aaaah..."
+}
 
 function onUse(player, item, fromPosition, target, toPosition, isHotkey)
-	local targetItemType = ItemType(target.itemid)
-	if targetItemType and targetItemType:isFluidContainer() then
+	local targetType = ItemType(target.itemid)
+	if targetType:isFluidContainer() then
```
Comment (WibbenZ) [inline data/actions/scripts/other/fluids.lua]: Why did you remove "if targetItem " - what if the target is not an item?

File: data/actions/scripts/other/fluids.lua [hunk 2]
```diff
@@ -26,37 +36,38 @@ function onUse(player, item, fromPosition, target, toPosition, isHotkey)
...
 		if item.type == 0 then
 			player:sendTextMessage(MESSAGE_STATUS_SMALL, "It is empty.")
-		elseif target == player then
-			item:transform(item:getId(), 0)
-			if item.type == 3 or item.type == 15 then
+
+		elseif target.uid == player.uid then
+			if table.contains({3, 15, 43}, item.type) then
 				player:addCondition(drunk)
+
```
Comment (WibbenZ) [inline data/actions/scripts/other/fluids.lua]: Remove the new line

File: data/actions/scripts/other/fluids.lua [hunk 3]
```diff
@@ -26,37 +36,38 @@ function onUse(player, item, fromPosition, target, toPosition, isHotkey)
...
 	end
 
-	if target:isCreature() then
+	if target.itemid == 1 then
 		if item.type == 0 then
 			player:sendTextMessage(MESSAGE_STATUS_SMALL, "It is empty.")
-		elseif target == player then
-			item:transform(item:getId(), 0)
-			if item.type == 3 or item.type == 15 then
+
```
Comment (WibbenZ) [inline data/actions/scripts/other/fluids.lua]: Remove the new line

File: data/actions/scripts/other/fluids.lua [hunk 4]
```diff
@@ -26,37 +36,38 @@ function onUse(player, item, fromPosition, target, toPosition, isHotkey)
...
+		elseif target.uid == player.uid then
+			if table.contains({3, 15, 43}, item.type) then
 				player:addCondition(drunk)
+
 			elseif item.type == 4 then
 				player:addCondition(poison)
+
```
Comment (WibbenZ) [inline data/actions/scripts/other/fluids.lua]: Remove the new line

File: data/actions/scripts/other/fluids.lua [hunk 5]
```diff
@@ -26,37 +36,38 @@ function onUse(player, item, fromPosition, target, toPosition, isHotkey)
...
 			elseif item.type == 7 then
 				player:addMana(math.random(50, 150))
 				fromPosition:sendMagicEffect(CONST_ME_MAGIC_BLUE)
+
```
Comment (WibbenZ) [inline data/actions/scripts/other/fluids.lua]: Remove the new line

File: data/actions/scripts/other/fluids.lua [hunk 6]
```diff
@@ -26,37 +36,38 @@ function onUse(player, item, fromPosition, target, toPosition, isHotkey)
...
+			player:say(fluidMessage[item.type] or "Gulp.", TALKTYPE_MONSTER_SAY)
+			item:transform(item:getId(), 0)
 		else
 			Game.createItem(2016, item.type, toPosition):decay()
 			item:transform(item:getId(), 0)
 		end
 	else
-		local fluidSource = targetItemType and targetItemType:getFluidSource() or 0
+
```
Comment (WibbenZ) [inline data/actions/scripts/other/fluids.lua]: Remove the new line

File: data/actions/scripts/other/fluids.lua [hunk 7]
```diff
@@ -26,37 +36,38 @@ function onUse(player, item, fromPosition, target, toPosition, isHotkey)
...
+		local fluidSource = targetType:getFluidSource()
 		if fluidSource ~= 0 then
 			item:transform(item:getId(), fluidSource)
+
```
Comment (WibbenZ) [inline data/actions/scripts/other/fluids.lua]: Remove the new line

File: data/actions/scripts/other/fluids.lua (hunk 1 above)
Comment (Dagst) [reply data/actions/scripts/other/fluids.lua]: Sorry, it was a mistake

File: data/actions/scripts/other/fluids.lua [hunk 8]
```diff
@@ -42,19 +51,14 @@ function onUse(player, item, fromPosition, target, toPosition, isHotkey)
...
-			player:say("Gulp.", TALKTYPE_MONSTER_SAY)
+			player:say(fluidMessage[item.type] or "Gulp.", TALKTYPE_MONSTER_SAY)
+			item:transform(item:getId(), 0)
 		else
 			Game.createItem(2016, item.type, toPosition):decay()
 			item:transform(item:getId(), 0)
 		end
 	else
-		local fluidSource = targetItemType and targetItemType:getFluidSource() or 0
+		local fluidSource = targetType:getFluidSource()
```
Comment (WibbenZ) [inline data/actions/scripts/other/fluids.lua]: targetType should be targetItemType (nil value)
